R/utils.R

```diff
@@ -177,20 +177,31 @@ generateResultFileNames <- function(numberOfCores, folderName, fileName, separat
 #' @return simulation object with pathIDs updated from simulationSet
 #' @export
 loadSimulationWithUpdatedPaths <- function(simulationSet, loadFromCache = FALSE) {
-  sim <- ospsuite::loadSimulation(
+  simulation <- ospsuite::loadSimulation(
     filePath = simulationSet$simulationFile,
     loadFromCache = loadFromCache,
     addToCache = FALSE
   )
   # Prevent loadSimulationWithUpdatedPaths from crashing if user did not submit any pathID
   if (!is.null(simulationSet$outputs)) {
-    sim$outputSelections$clear()
+    simulation$outputSelections$clear()
     paths <- sapply(simulationSet$outputs, function(output) {
       output$path
     })
-    ospsuite::addOutputs(quantitiesOrPaths = paths, simulation = sim)
+    ospsuite::addOutputs(quantitiesOrPaths = paths, simulation = simulation)
   }
-  return(sim)
+
+  if (is.null(simulationSet$minimumSimulationEndTime)) {
+    return(simulation)
+  }
+
+  if (simulationSet$minimumSimulationEndTime > simulation$outputSchema$endTime) {
+    maximalIntervalIndex <- which(sapply(simulation$outputSchema$intervals, function(x) {
+      x$endTime$value
+    }) == simulation$outputSchema$endTime)[1]

```

Comment:
why [1] here? Is this obvious?

Reply:
I think it depends on whether there could be multiple intervals in the outputSchema that have the same end time.  If not, then the [1] can be removed.